File: java/TopicInnerClasses.java

public class TopicInnerClasses
{

 public static void main( String[] args)
 {
  TestEnv te= new TestEnv();

  te.checkComplainAndAdjustExpected( 0);

  class Outer
  {
   int i;
   public Outer( int i){ this.i= i;}

   class Inner
   {
    int i;
    public Inner() { this.i= Outer.this.i;}
   }
  }

  Outer o= new Outer( 456);
  Outer.Inner i= o. new Inner(); 

  te.test( 456 == i.i);

  te.checkComplainAndAdjustExpected( 1);

  System.out.println( ""+ te.result());
 }
}

File: python-2.6.4/topic-inner-classes.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# vi:encoding=utf-8

from TestEnv import TestEnv

te= TestEnv()

te.checkComplainAndAdjustExpected( 0)

class A:
 def __init__( self):
  A_self=self
  class B:
   def getOuter( self):
    return A_self
  A_self.B=B

a1=A()
b11=a1.B()
b12=a1.B()

a2=A()
b21=a2.B()
b22=a2.B()

te.test( a1== b11.getOuter())
te.test( a1== b12.getOuter())

te.test( a2== b21.getOuter())
te.test( a2== b22.getOuter())

te.checkComplainAndAdjustExpected( 4)

# see also: 

print te.result()
